test/streams.js

```diff
@@ -3,34 +3,35 @@ var test = require('tape');
 
 test('pipe a thousand msgs between incompatible socket types', function(t){
 
+  t.plan(1);
+
   var sent = 0, recv = 0;
 
-  var pub = nano.socket('pub', { tcpnodelay: true });
-  var sub = nano.socket('sub', { tcpnodelay: true });
+  var pub = nano.socket('pub');
+  var sub = nano.socket('sub');
 
   var push = nano.socket('push');
   var pull = nano.socket('pull');
   pull.setEncoding('utf8'); //should also be able to do in `socket(type, opts)`
 
-  pub.bind('tcp://127.0.0.1:64999');
-  sub.connect('tcp://127.0.0.1:64999');
+  pub.bind('tcp://127.0.0.1:34999');
+  sub.connect('tcp://127.0.0.1:34999');
 
-  pull.bind('tcp://127.0.0.1:65000');
-  push.connect('tcp://127.0.0.1:65000');
+  pull.bind('tcp://127.0.0.1:35000');
+  push.connect('tcp://127.0.0.1:35000');
```

Comment:
yea I think so, we need more unique ports. This could all be a red herring: probably just the test running right before stream causing idiosyncratic timeouts. 

but it's cool we're improving the `socket.close()` implementation :+1: